helmtemplate/helm.go

```diff
@@ -66,15 +86,47 @@ func (t TemplateHelmChartTask) Run() error {
 			return microerror.Mask(err)
 		}
 
+		if validate && ValidateChart(buildInfo.Version, buildInfo.AppVersion, buf) != nil {
+			return microerror.Mask(err)
+		}
+
 		if err := afero.WriteFile(t.fs, path, buf.Bytes(), permission); err != nil {
 			return microerror.Mask(err)
 		}
 	}
 	return nil
 }
 
+// ValidateChart makes sure version fields and values made it into the
+// chart when executing the template.
+func ValidateChart(version, appVersion string, chartBuf bytes.Buffer) error {
+	chart := renderedChart{}
+	err := yaml.Unmarshal(chartBuf.Bytes(), &chart)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if chart.Version != version {
+		return microerror.Maskf(
+			versionValidationError,
+			"validation error: chart does not contain \"version\": %q", version,
```

Comment:
No need to repeat `validation error` as its already contained in `versionValidationError` error.

```suggestion
			"wrong value for \"version\" in chart: got %#q, expect %#q", chart.Version, version,
```